### PR TITLE
python3Packages.cvxpy: 1.1.11 -> 1.1.12

### PR DESCRIPTION
**Pull request by drewrisinger**
###### Motivation for this change
https://github.com/cvxgrp/cvxpy/compare/v1.1.11...v1.1.12

Add support for openmp multithreading when solving convex optimization expressions.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119225` at 286b532c run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/4bf1b21010e6536bb18f2934ee160077">python38Packages.cvxpy</a></li>
    <li><a href="https://gist.github.com/4b7b7f3b3a38ca3c45f6e28d1e99a464">python38Packages.qiskit</a></li>
    <li><a href="https://gist.github.com/5d801ad4c3e91ebd3aa36bf0e6f0ace1">python38Packages.qiskit-aer</a></li>
    <li><a href="https://gist.github.com/df79dff668ee7e39c74c3c6481c8bbde">python38Packages.qiskit-aqua</a></li>
    <li><a href="https://gist.github.com/11a3f0822ec9b579eef816806852b8b0">python38Packages.qiskit-ibmq-provider</a></li>
    <li><a href="https://gist.github.com/2905faf3ad4992e17b3aebb6944f9a68">python38Packages.qiskit-ignis</a></li>
    <li><a href="https://gist.github.com/5d598f6282b2e96815bf663455dde02f">python39Packages.cvxpy</a></li>
    <li><a href="https://gist.github.com/a6cfdaadb3423d900cd975b3575669d7">python39Packages.qiskit</a></li>
    <li><a href="https://gist.github.com/1059df411b8bcf55714cf5dbdd047644">python39Packages.qiskit-aer</a></li>
    <li><a href="https://gist.github.com/9b2cd7e87627016f46d91fffea51d283">python39Packages.qiskit-aqua</a></li>
    <li><a href="https://gist.github.com/33ba72188df19a5b14b9e8c3b3c18338">python39Packages.qiskit-ibmq-provider</a></li>
    <li><a href="https://gist.github.com/1d11802c45db47ec123cb92f661769be">python39Packages.qiskit-ignis</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

**Comment by rmcgibbo** [comment]
r-rmcgibbo failure on aarch64 is pre-existing.

It looks like they recently added some OpenMP based parallelization. We may want to update the expression to compile that in? https://github.com/cvxgrp/cvxpy/releases/tag/v1.1.11

**Comment by rmcgibbo** [comment]
I think on balance given the response from upstream (https://github.com/cvxgrp/cvxpy/issues/1241#issuecomment-780912155) we should disable the ECOS_BB test that's failing in cvxpy.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 119225` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>python39Packages.qiskit: log was empty</li>
<li>python39Packages.qiskit-aqua: log was empty</li>
  </ul>
</details>
<details>
<summary>
10 packages built:
</summary>
<ul>
<li>python38Packages.cvxpy</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-aer</li>
<li>python38Packages.qiskit-aqua</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.qiskit-ignis</li>
<li>python39Packages.cvxpy</li>
<li>python39Packages.qiskit-aer</li>
<li>python39Packages.qiskit-ibmq-provider</li>
<li>python39Packages.qiskit-ignis</li>
</ul>
</details>

**Comment by drewrisinger** [comment]
> 
> 
> I think on balance given the response from upstream ([cvxgrp/cvxpy#1241 (comment)](https://github.com/cvxgrp/cvxpy/issues/1241#issuecomment-780912155)) we should disable the ECOS_BB test that's failing on aarch64 in cvxpy.

Done in the last force-push.

> It looks like they recently added some OpenMP based parallelization. We may want to update the expression to compile that in? https://github.com/cvxgrp/cvxpy/releases/tag/v1.1.11

I have no objection to doing this, but it'll take some more time. How about I prototype that in my NUR repo, and upstream it to ``nixpkgs`` once I get the OpenMP parallelization working in a separate PR?

**Comment by drewrisinger** [comment]
@rmcgibbo OpenMP parallelism WIP: https://github.com/drewrisinger/nur-packages/pull/89

**Review by rmcgibbo — APPROVED** [review]

**Comment by drewrisinger** [comment]
@rmcgibbo It was easier than expected to add OpenMP, folded into this PR.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -1652,7 +1652,9 @@ in {
 
   cvxopt = callPackage ../development/python-modules/cvxopt { };
 
-  cvxpy = callPackage ../development/python-modules/cvxpy { };
+  cvxpy = callPackage ../development/python-modules/cvxpy {
+    inherit (pkgs.llvmPackages) openmp;
```
Comment (rmcgibbo) [inline pkgs/top-level/python-packages.nix:1656]: Does this mean that we're using the clang OpenMP even if the package is built with gcc?

Comment (drewrisinger) [reply pkgs/top-level/python-packages.nix:1656]: Ah, didn't know there was one built-in to GCC. Thanks, I'll fix that.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 119225` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
12 packages built:
</summary>
<ul>
<li>python38Packages.cvxpy</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-aer</li>
<li>python38Packages.qiskit-aqua</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.qiskit-ignis</li>
<li>python39Packages.cvxpy</li>
<li>python39Packages.qiskit</li>
<li>python39Packages.qiskit-aer</li>
<li>python39Packages.qiskit-aqua</li>
<li>python39Packages.qiskit-ibmq-provider</li>
<li>python39Packages.qiskit-ignis</li>
</ul>
</details>

**Pull request merged**